examples/Remarketing/UploadStoreSalesTransactions.cs

```diff
@@ -0,0 +1,488 @@
+
+            // Optional (but recommended) external ID to identify the offline user data job.
+            long? externalId = long.Parse("INSERT_EXTERNAL_ID_HERE");
+
+            // OPTIONAL: If uploading third party data, also specify the following values:
+            string advertiserUploadDateTime = "INSERT_ADVERTISER_UPLOAD_DATE_TIME_HERE";
+            string bridgeMapVersionId = "INSERT_BRIDGE_MAP_VERSION_ID_HERE";
+            long? partnerId = long.Parse("INSERT_PARTNER_ID_HERE");
+
+
+            codeExample.Run(new GoogleAdsClient(), customerId, conversionActionId,
+                offlineUserDataJobType: offlineUserDataJobType,
+                externalId: externalId,
+                advertiserUploadDateTime: advertiserUploadDateTime,
+                bridgeMapVersionId: bridgeMapVersionId,
+                partnerId: partnerId);
+        }
+
+        /// <summary>
+        /// Returns a description about the code example.
+        /// </summary>
+        public override string Description
+        {
+            get
+            {
+                return "This code example uploads offline data for store sales transactions. " +
+                       "This feature is only available to allowlisted accounts. " +
+                       "See https://support.google.com/google-ads/answer/7620302 for more details.";
+            }
+        }
+
+        /// <summary>
+        /// Runs the code example.
+        /// </summary>
+        /// <param name="client">The Google Ads client.</param>
+        /// <param name="customerId">The Google Ads customer ID for which the call is made.</param>
+        /// <param name="conversionActionId">The ID of a store sales conversion action.</param>
+        /// <param name="offlineUserDataJobType">The type of user data in the job (first or third
+        ///     party). If you have an official store sales partnership with Google, use
+        ///     StoreSalesUploadThirdParty. Otherwise, use StoreSalesUploadFirstParty or
+        ///     omit this parameter.</param>
+        /// <param name="externalId">Optional (but recommended) external ID to identify the offline
+        ///     user data job.</param>
+        /// <param name="advertiserUploadDateTime">Date and time the advertiser uploaded data to the
+        ///     partner. Only required if uploading third party data.</param>
+        /// <param name="bridgeMapVersionId">Version of partner IDs to be used for uploads. Only
+        ///     required if uploading third party data.</param>
+        /// <param name="partnerId">ID of the third party partner. Only required if uploading third
+        ///     party data.</param>
+        public void Run(GoogleAdsClient client, long customerId, long conversionActionId,
+            OfflineUserDataJobType offlineUserDataJobType =
+                OfflineUserDataJobType.StoreSalesUploadFirstParty,
+            long? externalId = null, string advertiserUploadDateTime = null,
```

Comment:
You don't need default values for strings or long?s. Feel free to remove it for everything other than the offlineUserDataJobType  parameter.

Reply:
These are intended to be optional parameters, though.  They're only required if using the Third Party job type; I could overload the Run method to have a different signature instead?

Reply:
On second thought the example runner probably won't support that idea. The only method that needs these is CreateOfflineUserDataJob. I feel like we'd like to avoid having to pass a bunch of null parameters in one of the two job types.